test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -2046,4 +2046,124 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
       });
     });
   });
+
+  context('16. Rewrap', function () {
+    const masterKeys = {
+      aws: {
+        region: 'us-east-1',
+        key: 'arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0'
+      },
+      azure: {
+        keyVaultEndpoint: 'key-vault-csfle.vault.azure.net',
+        keyName: 'key-name-csfle'
+      },
+      gcp: {
+        projectId: 'devprod-drivers',
+        location: 'global',
+        keyRing: 'key-ring-csfle',
+        keyName: 'key-name-csfle'
+      },
+      kmip: {},
+      local: undefined
+    };
+    let client1, client2;
+
+    /**
+     * Run the following test case for each pair of KMS providers (referred to as ``srcProvider`` and ``dstProvider``).
+     * Include pairs where ``srcProvider`` equals ``dstProvider``.
+     */
+    function* generateTestCombinations() {
+      const providers = Object.keys(masterKeys);
+      for (const srcProvider of providers) {
+        for (const dstProvider of providers) {
+          yield { srcProvider, dstProvider };
+        }
+      }
+    }
+
+    beforeEach(async function () {
+      client1 = await this.configuration.newClient().connect();
+      client2 = await this.configuration.newClient().connect();
+    });
+
+    afterEach(async function () {
+      await client1.close();
+      await client2.close();
+    });
+
+    for (const { srcProvider, dstProvider } of generateTestCombinations()) {
```

Comment:
This is a great pattern that I’ll absolutely keep in mind – it’s much nicer than nesting for loops around the test implementation :slightly_smiling_face:

Reply:
🙂